lib/ansible/executor/task_executor.py

```diff
@@ -507,7 +507,8 @@ def _execute(self, variables=None):
 
         if self._task.delegate_to:
             # use vars from delegated host (which already include task vars) instead of original host
-            cvars = variables.get('ansible_delegated_vars', {}).get(self._task.delegate_to, {})
+            delegate_vars = variables.get('ansible_delegated_vars', {}).get(self._task.delegate_to, {})
+            cvars = combine_vars(variables, delegate_vars)
```

Comment:
no, combine vars with inventory_hostname variables is specifically what we want to avoid, this does not only bring the `-k` defined password but any that where set in inventory for the inventory_hostname, then applying possibly wrong password (among other settings) to the delegated_to host.  What we do need to do is pull the play_context passwords  (in the end they should really be applied to the task, but that is future fix) and let delegated_to host override if set.

Reply:
need to x2 check, but `  self._play_context.update_vars(cvars)` shoudl do the trick here

Reply:
currently other plugins use play_context directly as a fallback ` conn_password = self.get_option('password') or self._play_context.password`

Reply:
after thinking for a bit, bets place should be to add passwords to the `set_options` calls in  task_executor, both for connection and become by merging into 'task_keys' before passing it in (at lowest priority) which will allow settings/vars to keep the right precedence and insure its always present

Reply:
https://github.com/ansible/ansible/blob/devel/lib/ansible/executor/task_executor.py#L903

Reply:
actually .. it seems we do attempt to do that .. could it be we have a bug in that code?

Reply:
ah ... winrm uses `remote_password` not `password` .. why it is not getting picked up.

Reply:
not sure how we want to fix it, add 'remote_password' as another task field or update the winrm plugin?

Reply:
may I ask please why WinRM uses remote_password and not password ?

Reply:
@samalv they are just keyword names for the config options which they never really were used for anything special before. I mostly followed the same standard for `remote_user` and `remote_addr` but after this change it does have a meaning. I've added the alias for `password` and fixed the code to actually use the aliases when setting the options for keywords.

Reply:
@jborean93 thanks for explanation, should we expect this fix to be pushed in next Ansible release 2.9.12 ? or shall I do the same fix manually on my installed version 2.9.11 ?

Reply:
I'll be creating a backport for this so the next 2.9 release will include it. Unfortunately it is too late for 2.10.0 so it will be in the 2.10.1 release later on.

Reply:
yes I saw them, what about 2.11 ?